### PR TITLE
Clippy lints fixes for latest code 

### PR DESCRIPTION
**Pull request by makr11st**
## Proposed changes

clippy lints mainly:
* remove redundant references 
* remove unnecessary `clone()` calls
* redundant `Error` in error variants names

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: mapper/tedge_mapper/src/collectd_mapper/error.rs [hunk 1]
```diff
@@ -3,19 +3,17 @@ use tokio::sync::mpsc::error::SendError;
 #[derive(thiserror::Error, Debug)]
 pub enum DeviceMonitorError {
     #[error(transparent)]
-    MqttClientError(#[from] mqtt_client::MqttClientError),
+    MqttClient(#[from] mqtt_client::MqttClientError),
```
Comment (albinsuresh) [inline mapper/tedge_mapper/src/collectd_mapper/error.rs:6]: That's a really weird name for an error. The "failure" aspect is missing from the name. I personally prefer the earlier names with `Error` suffixes. Probably not for the errors that already has another negative term in them like "Invalid".

**Review by albinsuresh — APPROVED** [review]
Not a great fan of the renames done to the error variants removing the suffixes. An error named `MqttClient` without any prefix or suffix looks weird. For names like `InvalidJson` it makes sense as the negativity aspect is already captured in that name with the word "invalid'. But for those names that don't have that negativity term in it, I personally prefer the `Error` suffix. But, it's a subjective matter. If this is the standard way of naming errors in Rust or if the rest of the reviewers also prefer this, I'm okay with it.

File: mapper/tedge_mapper/src/collectd_mapper/error.rs (hunk 1 above)
Comment (rina23q) [reply mapper/tedge_mapper/src/collectd_mapper/error.rs:6]: I checked https://doc.rust-lang.org/src/std/io/error.rs.html, all error names are negative (e.g. TooManyLinks).
I think we had a discussion while implementing sm-mapper, and we decided to prefix `From`. See https://github.com/thin-edge/thin-edge.io/blob/main/mapper/tedge_mapper/src/sm_c8y_mapper/error.rs#L54. Here the name is `FromMqttClient`.
I prefer `FromMqttClient`.

File: mapper/tedge_mapper/src/collectd_mapper/error.rs [hunk 2]
```diff
@@ -3,19 +3,17 @@ use tokio::sync::mpsc::error::SendError;
...
 
     #[error(transparent)]
-    InvalidCollectdMeasurementError(#[from] crate::collectd_mapper::collectd::CollectdError),
+    InvalidCollectdMeasurement(#[from] crate::collectd_mapper::collectd::CollectdError),
 
     #[error(transparent)]
-    InvalidThinEdgeJsonError(#[from] thin_edge_json::group::MeasurementGrouperError),
+    InvalidThinEdgeJson(#[from] thin_edge_json::group::MeasurementGrouperError),
```
Comment (didier-wenzek) [inline mapper/tedge_mapper/src/collectd_mapper/error.rs:12]: Here the new name makes sense

File: mapper/tedge_mapper/src/collectd_mapper/error.rs [hunk 3]
```diff
@@ -3,19 +3,17 @@ use tokio::sync::mpsc::error::SendError;
...
 
     #[error(transparent)]
-    ThinEdgeJsonSerializationError(
-        #[from] thin_edge_json::serialize::ThinEdgeJsonSerializationError,
-    ),
+    ThinEdgeJsonSerialization(#[from] thin_edge_json::serialize::ThinEdgeJsonSerializationError),
 
     #[error(transparent)]
-    BatchingError(#[from] SendError<thin_edge_json::group::MeasurementGrouper>),
+    Batching(#[from] SendError<thin_edge_json::group::MeasurementGrouper>),
```
Comment (didier-wenzek) [inline mapper/tedge_mapper/src/collectd_mapper/error.rs:18]: But here, the former name is clearer.

**Review by didier-wenzek — COMMENTED** [review]
Looks good except for the error names. I would prefer to cherry pick these names rather than letting a tool format them all.

**Review by rina23q — APPROVED** [review]
I approve. There's a room to improve and fix our error naming patterns for transparent ones.

**Pull request merged**